File: sqlparser/src/sql/csgen/SqlSubQueryExpression.cs
using System;
using System.Collections;
using System.IO;
using System.Threading;
using System.Net;
using System.Net.Sockets;
using System.Text;
using RSSBus.core.j2cs;



using RSSBus.core;
using RSSBus;
namespace CData.Sql {


public sealed class SqlSubQueryExpression : SqlExpression{
  private readonly SqlQueryStatement query;

  public SqlSubQueryExpression(SqlQueryStatement subQuery) {
    this.query = subQuery;
  }

  public SqlQueryStatement GetQuery() {
    return this.query;
  }

  public override bool IsEvaluatable() {
    return false;
  }

  public override SqlValue Evaluate() {
    //Do nothing now.
    return SqlValue.GetNullValueInstance();
  }

  public override Object Clone() {
    return new SqlSubQueryExpression((SqlQueryStatement)query.Clone());
  }

  protected override void Copy(SqlExpression obj) {
  }
}
}

File: sqlparser/src/sql/SqlSubQueryExpression.java
//@
package cdata.sql;
//@
/*#
using RSSBus.core;
using RSSBus;
namespace CData.Sql {
#*/

public final class SqlSubQueryExpression extends SqlExpression{
  private final SqlQueryStatement query;

  public SqlSubQueryExpression(SqlQueryStatement subQuery) {
    this.query = subQuery;
  }

  public SqlQueryStatement getQuery() {
    return this.query;
  }

  public /*#override#*/ boolean isEvaluatable() {
    return false;
  }

  public /*#override#*/ SqlValue evaluate() {
    //Do nothing now.
    return SqlValue.getNullValueInstance();
  }

  @Override
  public Object clone() {
    return new SqlSubQueryExpression((SqlQueryStatement)query.clone());
  }

  @Override
  protected void copy(SqlExpression obj) {
  }
}
